File: manage-grp.Server/Models/Address.cs
using manage_grp.Server.Helpers;
using System.Text.Json.Serialization;

namespace manage_grp.Server.Models
{
    public class Address
    {
        public int? Id { get; set; }

        public int? DependencyId { get; set; }

        [JsonIgnore]
        public Dependency? Dependency { get; set; }

        public string Neighborhood { get; set; }

        public string Street { get; set; }

        public int ExteriorNumber { get; set; }

        public int? InteriorNumber { get; set; }

        public string PostalCode { get; set; }

        public string Latitude { get; set; }

        public string Longitude { get; set; }

        public string? Reference { get; set; }

        public bool IsActive { get; set; }

        public DateTime CreatedAt { get; set; } = DateHelper.GetTimeInTimeZone();

        public DateTime UpdatedAt { get; set; } = DateHelper.GetTimeInTimeZone();
    }
}

File: manage-grp.client/src/stores/app/address.ts
import axios from "axios";
import { defineStore } from "pinia";
import { error } from "@/utils/toast/Error";
import { success } from "@/utils/toast/Success";

interface IAddressValues {
  id?: number;
  dependencyId: number;
  neighborhood: string;
  street: string;
  exteriorNumber: string;
  interiorNumber: string;
  postalCode: string;
  latitude: string;
  longitude: string;
  reference: string;
  isActive: boolean;
}

export const useAddressStore = defineStore('address', () => {


  const storeOrUpdateAddress = async (values: IAddressValues, loading:{value: boolean}, t: any, isAddressStore: boolean) => {
    loading.value = true;

    try {
      const response = isAddressStore ? await axios.post('/api/Address', values)
                        : await axios.put(`/api/Address/${values.id}`, values);

      if(response.status === 200){
        console.log(response.data);
        success(t(response.data.message));
        return;
      }

      error(t('ERROR_MESSAGE'));
      return [];
    } catch (ex: any) {
      error(ex.response.data.message || t('ERROR_MESSAGE'));
    }
    finally {
      loading.value = false;
    }
  }

  return {
    storeOrUpdateAddress,
  }
})
